### PR TITLE
fix: remove runtime dependency on `load-rules`

### PR DESCRIPTION
**Pull request by llllvvuu**
dynamic loading/requiring of files breaks bundlers as those files will not be available at runtime of bundled code (#461)

i accomplish this by adding a script, `npm run generate-rulesets`, which generates a static version of the rulesets

i also add prepublish and precommit hook to make sure docs and rulesets are up-to-date if anything in the rules directory has changed

### CODE REVIEW

**Review by dbale-altoros — COMMENTED** [review]
why is this file `solhint.js` empty ?

File: package.json [hunk 1]
```diff
@@ -20,7 +20,10 @@
     "test:coverage": "npm run exec-tests",
     "test": "mocha --recursive",
     "lint": "eslint .",
-    "docs": "node scripts/generate-rule-docs.js"
+    "generate-rulesets": "node scripts/generate-rulesets.js && prettier --write conf/rulesets",
+    "docs": "node scripts/generate-rule-docs.js",
+    "prepare": "husky install",
```
Comment (dbale-altoros) [inline package.json:25]: Sorry if it is a dumb question, but
are the 'prepare' and 'preoublishOnly' needed ?

Comment (llllvvuu) [reply package.json:25]: It's not strictly necessary, but it makes docs and ruleset JSON automatically updated when you commit or publish

**Comment by llllvvuu** [comment]
`solhint.js` was a mistake, removed

File: package.json (hunk 1 above)
Comment (llllvvuu) [reply package.json:25]: Oh, I see what you're asking.

Those are [`npm` lifecycle scripts](https://docs.npmjs.com/cli/v9/using-npm/scripts#life-cycle-scripts). `npm` will run `prepare` when you run `npm install`, that makes sure that `git` is synced with `husky`.

`npm` will run `prepublishOnly` when you run `npm publish`, that makes sure that `npm` package has up to date rulesets.

**Pull request merged**